lib/p2p/Pool.ts

```diff
@@ -183,6 +183,25 @@ class Pool extends EventEmitter {
    * Attempt to create an outbound connection to a node using its known listening addresses.
    */
   private tryConnectNode = async (node: NodeConnectionInfo, retryConnecting = false) => {
+    if (!await this.connectWithLastAddress(node)) {
+      await this.connectWithAdvertisedAddresses(node, retryConnecting);
+    }
+  }
+
+  private connectWithLastAddress = async (node: NodeConnectionInfo) => {
+    const { lastAddress, nodePubKey } = node;
+
+    if (!lastAddress) return false;
+
+    try {
+      await this.addOutbound(lastAddress, nodePubKey, false);
+      return true;
+    } catch (err) {}
+
+    return false;
+  }
+
+  private connectWithAdvertisedAddresses = async (node: NodeConnectionInfo, retryConnecting = false) => {
     const { addresses, nodePubKey } = node;
```

Comment:
This method contain the old logic, where retries are applies on the last connected address from the advertised addresses. As @sangaman mentioned, we should apply the retry logic on `lastAddress` instead - which might not have been advertised. It also means that we should extract the retries section from here to a different method, to be called from  `tryConnectNode`.

In addition, there might be a case where `lastAddress` was advertised, so you will try connecting to it twice - once in `connectWithLastAddress`, and then again in `connectWithAdvertisedAddresses`. This wouldn't have been if we saved `lastAddress` solely for an unadvertised address. There are other ways to solve this as well. I'll leave it up to you, let us know if you need help with it.

Reply:
But if I put retry logic to the connecting last address part, wouldn't that prevent connecting to the advertised addresses ? Is this what we want ? 

I'll handle connecting twice issue by checking the address in connecting to advertised addresses method, if the address equals with lastAddress then I won't try to connect it in advertised address method.

Reply:
And wouldn't it be sufficient to send `retryConnecting = true` for `addOutbound` method inside of `tryConnectWithLastAddress` method

Reply:
Would you please check current implementation ?